src/addressfinder/reverseGeocode.tsx

```diff
@@ -0,0 +1,41 @@
+const addressFinderAPIKey = "ARFHPVK67QXM49BEWDL3";
+const reverseGeocodeURL = `https://api.addressfinder.io/api/nz/address/reverse_geocode/?key=${addressFinderAPIKey}&format=json`;
+
+export interface reverseGeocodeResp {
+  completions: addressResp[];
+  paid: boolean;
+  demo: boolean;
+  success: boolean;
+}
+
+interface addressResp {
+  a: string;
+  pxid: string;
+}
+
+// TODO(2021-09-13): handle error/edge cases better!
+async function getSuburb(lat: number, lng: number): Promise<string> {
+  try {
+    let resp = await fetch(reverseGeocodeURL + `&y=${lat}&x=${lng}`);
+    let data: reverseGeocodeResp = await resp.json();
+    if (data.completions.length === 0 || !data.success) {
+      return "";
+    }
+
+    return extractSuburb(data.completions[0].a);
+  } catch (err) {
+    throw new Error(`unable to reverse geocode coordinates: ${err}`);
+  }
+}
+
+function extractSuburb(addr: string): string {
+  let components = addr.split(",");
+
+  if (components.length === 0) {
+    return "";
+  }
+
+  return components[components.length - 2];
```

Comment:
addressfinder.nz only returns a single address string, instead of broken down address components. I'm making an assumption that the second to last component is the suburb based on addresses I've seen from their documentation and testing. Is there perhaps a better way to handle this?

Reply:
This is quite a big assumption that needs looking into. @noway can you put @rquitales in touch with the support team at AddressFinder.nz to work out a stable implementation of this as the array index presented here will topple over when we least expect it. We want to get something like the response [documented here](https://addressfinder.nz/api/nz/address/metadata/), where there is a clear suburb element returned. Maybe the Address Finder team can help.

Reply:
@olafwrieden Without talking to the Address Finder team, I think we can actually implement this and have the different address components by using the `pxid` from the reverse geocoder endpoint to then subsequently query the address metadata endpoint. The downside to this is obviously needing 2 API calls instead of 1.

Reply:
@rquitales Great idea! Let's do 2 api calls, that's perfect. I had to do something similar for Google Reverse Geocode too, so should be fine!

Reply:
Done.